src/c++/perf_analyzer/test_perf_utils.cc

```diff
@@ -114,4 +117,288 @@ TEST_CASE("testing the ParseTensorFormat function")
+}
+
+TEST_CASE("perf_utils: IsFile")
+{
+  // Create a temporary file /tmp/abc/test.txt
+  std::filesystem::path temp_path = std::filesystem::temp_directory_path();
+  temp_path /= "abc/test.txt";
+
+  CHECK(!IsFile(temp_path));
+
+  std::filesystem::create_directory(temp_path.parent_path());
+  std::ofstream file(temp_path);
+  CHECK(IsFile(temp_path));
+
+  std::filesystem::remove_all(temp_path.parent_path());
+  CHECK(!IsFile(temp_path));
+}
+
+TEST_CASE("perf_utils: ByteSize")
+{
+  std::vector<int64_t> shape{3, 4, 5};
+
+  SUBCASE("Single byte elements")
+  {
+    CHECK(ByteSize(shape, "BOOL") == 1 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "INT8") == 1 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT8") == 1 * 3 * 4 * 5);
+  }
+
+  SUBCASE("2 byte elements")
+  {
+    CHECK(ByteSize(shape, "INT16") == 2 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT16") == 2 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "FP16") == 2 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "BF16") == 2 * 3 * 4 * 5);
+  }
+
+  SUBCASE("4 byte elements")
+  {
+    CHECK(ByteSize(shape, "INT32") == 4 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT32") == 4 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "FP32") == 4 * 3 * 4 * 5);
+  }
+
+  SUBCASE("8 byte elements")
+  {
+    CHECK(ByteSize(shape, "INT64") == 8 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT64") == 8 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "FP64") == 8 * 3 * 4 * 5);
+  }
+
+  SUBCASE("Dynamic shape tensor")
+  {
+    shape.insert(shape.begin(), -1);
+
+    CHECK(ByteSize(shape, "BOOL") == -1);
+    CHECK(ByteSize(shape, "INT8") == -1);
+    CHECK(ByteSize(shape, "UINT8") == -1);
+
+    CHECK(ByteSize(shape, "INT16") == -1);
+    CHECK(ByteSize(shape, "UINT16") == -1);
+    CHECK(ByteSize(shape, "FP16") == -1);
+    CHECK(ByteSize(shape, "BF16") == -1);
+
+    CHECK(ByteSize(shape, "INT32") == -1);
+    CHECK(ByteSize(shape, "UINT32") == -1);
+    CHECK(ByteSize(shape, "FP32") == -1);
+
+    CHECK(ByteSize(shape, "INT64") == -1);
+    CHECK(ByteSize(shape, "UINT64") == -1);
+    CHECK(ByteSize(shape, "FP64") == -1);
+  }
+
+  SUBCASE("Unknown data types")
+  {
+    CHECK(ByteSize(shape, "bool") == -1);
+    CHECK(ByteSize(shape, "int8") == -1);
+    CHECK(ByteSize(shape, "uint8") == -1);
+
+    CHECK(ByteSize(shape, "int16") == -1);
+    CHECK(ByteSize(shape, "uint16") == -1);
+    CHECK(ByteSize(shape, "fp16") == -1);
+    CHECK(ByteSize(shape, "bf16") == -1);
+
+    CHECK(ByteSize(shape, "int32") == -1);
+    CHECK(ByteSize(shape, "uint32") == -1);
+    CHECK(ByteSize(shape, "fp32") == -1);
+
+    CHECK(ByteSize(shape, "int64") == -1);
+    CHECK(ByteSize(shape, "uint64") == -1);
+    CHECK(ByteSize(shape, "fp64") == -1);
+
+    CHECK(ByteSize(shape, "abc") == -1);
+    CHECK(ByteSize(shape, "1234") == -1);
+    CHECK(ByteSize(shape, "") == -1);
+  }
+}
+
+TEST_CASE("perf_utils: ElementCount")
+{
+  std::vector<int64_t> shape{3, 4, 5};
+
+  SUBCASE("Static tensor shape")
+  {
+    CHECK(ElementCount(shape) == 60);
```

Comment:
Show us the calculation or include it in a comment. It is the multiplication of all shape elements(?)